frontend/src/contexts/agreement.tsx

```diff
@@ -15,13 +16,46 @@ const AgreementProvider: FC = () => {
       agreementId,
     },
   });
+
+  const [voteFn, { error: voteError }] = useMutation(voteMutation, {
+    refetchQueries: ['votes'],
+  });
+
+  const onVote = async (user_id: number, version_id: number, type: 'up' | 'down') => {
+    if (!user_id || !version_id) {
+      console.error(
+        `error in voting: bad arguments: user_id: ${user_id}, version_id: ${version_id}`
+      );
+      return;
+    }
+    if (type !== 'up' && type !== 'down') {
+      console.error(`error in voting: vote can be type up or down`);
+      return;
+    }
+    const voteRes = await voteFn({
+      variables: {
+        user_id,
+        version_id,
+        type,
+      },
+    });
+    if (voteError) {
+      console.error(`error in voting: ${voteError}`);
+      return;
+    }
+    return voteRes;
+  };
+
   const agreement = data?.core_agreements[0];
   const state: IAgreementContext = {
     agreementId: agreement?.id || NaN,
     categoryName: agreement?.category?.name || '',
     rationale: agreement?.rationale || '',
     agreementTitle: agreement?.name || '',
     agreement: agreement,
+    vote: (user_id: number, version_id: number, type: 'up' | 'down') => {
+      return onVote(user_id, version_id, type);
```

Comment:
Please refactor this to be a one-liner

Reply:
its too long, so the linter turns it into multiple lines